Sources/Utility/ArgumentParser.swift

```diff
@@ -109,35 +121,22 @@ extension Int: ArgumentKind {
         self.init(arg)
     }
 
-    public init(parser: inout ArgumentParserProtocol) throws {
-        let arg = try parser.associatedArgumentValue ?? parser.next()
-        // Not every string can be converted into an integer.
-        guard let intValue = Int(arg) else {
-            throw ArgumentParserError.typeMismatch("\(arg) is not convertible to Int")
-        }
-        self = intValue
-    }
-
     public static let completion: ShellCompletion = .none
 }
 
 extension Bool: ArgumentKind {
     public init?(arg: String) {
-        self = true
+        switch arg {
```

Comment:
iirc this will never be called for Bool argument, do we have test coverage for this? We should just `fatalError` here if this doesn't apply for Bools.